RoddyCore/src/de/dkfz/roddy/core/CohortDataRuntimeServiceExtension.groovy

```diff
@@ -125,14 +136,94 @@ class CohortDataRuntimeServiceExtension {
 
             def cohortDataSets = superCohort.y.collect { String cohortID, List<String> cohortDatasetIdentifier ->
                 List<DataSet> dList = collectDataSetsForCohort(cohortDatasetIdentifier as String[], analysis, listOfDataSets)
-                dList = dList.sort { a,b -> a.id.compareTo(b.id) }.unique()
+                dList = dList.sort { a, b -> a.id.compareTo(b.id) }.unique()
                 return new CohortDataSet(analysis, cohortID, dList, new File(superCohortOutputDirectory, cohortID))
             }
             return new SuperCohortDataSet(analysis, superCohortID, cohortDataSets, superCohortOutputDirectory)
         }
         return datasets as List<DataSet>
     }
 
+    /**
+     * Returns true, if all filters are valid super cohort identifiers.
+     * @return
+     */
+    boolean checkIfListOnlyContainsSCIdentifiers(List<String> pidFilters) {
+        // We use the COHORT_IDENTIFIER. Rules for it also apply for the super cohort identifier.
+        pidFilters.every { it ==~ COHORT_IDENTIFIER }
+    }
+
+    /**
+     * 1. Check, if super cohort cvalue is present.
+     * 2. Check, if super cohort cvalue is valid.
+     * 3. Check for all cohort identifiers in a super cohort cvalue, if they are present.
+     * 4. Check for all cohort cvalues, if they are valid.
+     * @param configuration
+     * @param scIdentifiers
+     */
+    void validateSuperCohortCValuesOrFail(Configuration configuration, List<String> scIdentifiers) {
+        def values = configuration.configurationValues.allValues
+
+        // 1. Check, if super cohort cvalue is present.
```

Comment:
Why the comments in this long method? Better make small private methods whose name express what you comment here.

Reply:
That's also less daunting to review. As it is I expected complexity -- but the method is probably easy. That should be reflected by the code.